libevmasm/AssemblyItem.h

```diff
@@ -51,6 +51,7 @@ enum AssemblyItemType
 	PushDeployTimeAddress, ///< Push an address to be filled at deploy time. Should not be touched by the optimizer.
 	PushImmutable, ///< Push the currently unknown value of an immutable variable. The actual value will be filled in by the constructor.
 	AssignImmutable, ///< Assigns the current value on the stack to an immutable variable. Only valid during creation code.
+	DataLoadN, /// Loads 32 bytes from EOF data section. TODO: Why cannot it be done with builtin function?
```

Comment:
Question is if you need access from EVM level to the data section data added  by the `Assembly::assemble` in particular `Assembly::m_data` and the `Assembly::m_auxiliaryData`. Not sure what are all case when it's used. In Legacy EVM you were able to access it from EVM context it's possible for EOF too but it can be tricky to implement because when generating contract creation code in yul we have to take into consideration that there will be some additional data before and adjust the offsets accordingly.

Reply:
Regarding the comment it's the old one which should be removed now. I was wondering it it can be done easier w/o adding new `AssemblyItem` type